### PR TITLE
add TRUE/FALSE definitions to configure rules.

### PR DESCRIPTION
**Pull request by yoya**
I propose to add TRUE and FALSE definitions.
only boolean definition will confuse the jpeg header. 

* jmorecfg.h (jpeg9c)

```
#ifndef HAVE_BOOLEAN
#if defined FALSE || defined TRUE || defined QGLOBAL_H
/* Qt3 defines FALSE and TRUE as "const" variables in qglobal.h */
typedef int boolean;
#ifndef FALSE                   /* in case these macros already exist */
#define FALSE   0               /* values of boolean */
#endif
#ifndef TRUE
#define TRUE    1
#endif
#else
typedef enum { FALSE = 0, TRUE = 1 } boolean;
#endif
#endif
```

We will need to set the boolean type, TRUE and FALSE together in configure.

### CODE REVIEW

**Comment by matthiaskramm** [comment]
For my understanding: Does this fix a bug? Or is it more of a refactoring?

**Comment by yoya** [comment]
It's intended for bug fixes.
libjpeg header file(jmorecfg.h) defines boolean/TRUE/FALSE when there is no boolean environment, but swftools defines only boolean, which causes problems on platforms without TRUE and FALSE.

In particular, this cause a problem on macOS.

```
gcc -c -DHAVE_CONFIG_H   -I/Users/yoya/swftools/master/include -I/usr/local/opt/freetype/include/freetype2 -Ilame -Ilib/lame -fPIC -Wimplicit -Wreturn-type -Wno-write-strings -Wformat -O -fomit-frame-pointer  jpeg.c -o jpeg.o
jpeg.c:109:35: error: use of undeclared identifier 'TRUE'
  jpeg_set_quality(&cinfo,quality,TRUE);
                                  ^
jpeg.c:113:31: error: use of undeclared identifier 'FALSE'
  jpeg_start_compress(&cinfo, FALSE);
                              ^
jpeg.c:151:35: error: use of undeclared identifier 'TRUE'
  jpeg_set_quality(&cinfo,quality,TRUE)
(omit...)
```

**Comment by yoya** [comment]
this PR. I changed only configure.in manually.
config.h.in and configure were generated by the following command.

```
autoreconf -i -f -I m4
```

**Comment by yoya** [comment]
The first time I ran autoreconf on macOS Catalina(10.15.2) & homebrew autoconf 2.69
https://github.com/matthiaskramm/swftools/pull/67/commits/391c7d6c0c0ce756ab77692df72665c9b1f2f395

When I ran autoreconf again on Ubuntu18 & 2.69, the runstatedir rule was added.
https://github.com/matthiaskramm/swftools/pull/67/commits/36a3d6326e6cfdc4724f8f77012e3f0c28c2386c
Debian and Ubuntu seem to support runstatedir independently.

Consequently, this Pull Request support runstatedir.
If not needed, so I'll revert it.

**Comment by yoya** [comment]
This PR resolves the following issue:

- Compiling Errors: lib/jpeg.c needs update
   - https://github.com/matthiaskramm/swftools/issues/37

File: config.h.in [hunk 1]
```diff
@@ -261,6 +336,7 @@
 #ifdef HAVE_POPPLER
 #define GString GooString
 #define GHash GooHash
+#define GHashIter GooHashIter
```
Comment (matthiaskramm) [inline config.h.in]: What does this change have to do with the rest?

File: configure [hunk 2]
```diff
@@ -1,11 +1,9 @@
 #! /bin/sh
 # Guess values for system-dependent variables and create Makefiles.
-# Generated by GNU Autoconf 2.68.
+# Generated by GNU Autoconf 2.69.
```
Comment (matthiaskramm) [inline configure]: Could you remove the regenerated configure from the changelist? Either that, or generate it in a way that makes it more clear what changed (using autoconf 2.68 might help)

File: config.h.in (hunk 1 above)
Comment (yoya) [reply config.h.in]: This change has nothing to do with what we want to resolve in this issue.
One of the rules that autoreconf adds automatically.
The current config.h.in is unnatural as created by autotools.
I will undo if there is a reason config.h.in is so.

File: configure (hunk 2 above)
Comment (yoya) [reply configure]: I regenerated configure file with autoconf 2.68. (sorry commit comment typo 2.98)

File: config.h.in (hunk 1 above)
Comment (matthiaskramm) [reply config.h.in]: Yes, could you revert the config.h.in changes? This isn't an autogenerated file, so let's not make autotools reformat it.

Comment (yoya) [reply config.h.in]: Yes, I'm done. > revert 
https://github.com/matthiaskramm/swftools/pull/67/commits/9099735e340a1fa3b3034cdc1713c6d2fbb649c0

**Comment by matthiaskramm** [comment]
Looks good, merging. Do we need a follow-up PR to config.h.in to make the new constants visible? (Not talking about the formatting changes you just reverted, just the two-liner for true and false)

**Comment by yoya** [comment]
I'm sorry. Confirmation was halfway. There may be additional PR.
Please wait for a moment. I think merge is fine & can be left as is.

**Pull request merged**